kge/job/train.py

```diff
@@ -37,33 +38,27 @@ class TrainingJob(Job):
     """
 
     def __init__(
-        self, config: Config, dataset: Dataset, parent_job: Job = None
+        self, config: Config, dataset: Dataset, parent_job: Job = None, init=True
     ) -> None:
-        from kge.job import EvaluationJob
 
         super().__init__(config, dataset, parent_job)
-        self.model: KgeModel = KgeModel.create(config, dataset)
-        self.optimizer = KgeOptimizer.create(config, self.model)
-        self.kge_lr_scheduler = KgeLRScheduler(config, self.optimizer)
+        self.model: KgeModel
+        self.optimizer = None
+        self.kge_lr_scheduler = None
+        self.valid_job = None
         self.loss = KgeLoss.create(config)
         self.abort_on_nan: bool = config.get("train.abort_on_nan")
         self.batch_size: int = config.get("train.batch_size")
         self.device: str = self.config.get("job.device")
         self.train_split = config.get("train.split")
-        valid_conf = config.clone()
-        valid_conf.set("job.type", "eval")
-        if self.config.get("valid.split") != "":
-            valid_conf.set("eval.split", self.config.get("valid.split"))
-        valid_conf.set("eval.trace_level", self.config.get("valid.trace_level"))
-        self.valid_job = EvaluationJob.create(
-            valid_conf, dataset, parent_job=self, model=self.model
-        )
+
         self.config.check("train.trace_level", ["batch", "epoch"])
         self.trace_batch: bool = self.config.get("train.trace_level") == "batch"
         self.epoch: int = 0
         self.valid_trace: List[Dict[str, Any]] = []
         self.is_prepared = False
-        self.model.train()
+        if init:
+            self._initialize_training(config, dataset)
```

Comment:
Why is this not always done? An easier to read variant may be to pass "checkpoint" instead of init. The code can then decide what to do.

Reply:
removed